atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/metrics/SweepOutcomeMetricsTest.java

```diff
@@ -112,4 +113,23 @@ public void targetedSweepDoesNotRegisterExcludedOutcomes() {
                 .filter(outcome -> !SweepOutcomeMetrics.TARGETED_OUTCOMES.contains(outcome))
                 .forEach(outcome -> assertThat(metricsManager).hasNotRegisteredTargetedOutcome(THOROUGH, outcome));
     }
+
+    @Test
+    public void canFilterOutUninterestingMetrics() {
+        MetricsManager differentManager = MetricsManagers.createAlwaysSafeAndFilteringForTests();
+        SweepOutcomeMetrics.registerTargeted(
+                metricsManager,
+                ImmutableMap.of("strategy", "thorough"),
+                () -> false);
+        TargetedSweepMetrics targetedMetrics = TargetedSweepMetrics
```

Comment:
Sorry i din't understand why you need to register on SweepOutcomeMetrics at all/ with a different metrics manager?